request.js

```diff
@@ -1426,6 +1421,13 @@ Request.prototype.destroy = function () {
   }
 }
 
+Request.prototype.clearTimeout = function () {
+  if (this.timeout && this.timeoutTimer) {
```

Comment:
```diff
-  if (this.timeout && this.timeoutTimer) {
+  if (this.timeoutTimer) {
```

We should call `clearTimeout(this.timeoutTimer)` anytime `this.timeoutTimer` was created, even if `this.timeout` wasn't passed.